File: 백준/Silver/11047. 동전 0/동전 0.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let [n, ...input] = fs.readFileSync(filePath).toString().split("\n");

const moneyList = input.reverse();
let myMoney = parseInt(n.split(" ")[1]);
let moneyCnt = 0;

for (let i = 0; i < moneyList.length; i++) {
  if (myMoney >= parseInt(moneyList[i])) {
    let moneyCount = Math.floor(myMoney / parseInt(moneyList[i]));
    moneyCnt += Math.floor(myMoney / parseInt(moneyList[i]));
    myMoney -= parseInt(moneyList[i]) * moneyCount;
  }
}

console.log(moneyCnt);

File: 백준/Silver/11047. 동전 0/동전 0.py
import sys

input = sys.stdin.readline

N,M = map(int,input().split())
arr = sorted([int(input()) for _ in range(N)],reverse=True)
ans = 0

for i in range(len(arr)) :
  if arr[i] > M : continue
  ans += M // arr[i]
  M %= arr[i]
  
print(ans)